src/keyword_vpd_parser.cpp

```diff
@@ -0,0 +1,143 @@
+#include "keyword_vpd_parser.hpp"
+
+#include "constants.hpp"
+#include "exceptions.hpp"
+
+#include <iostream>
+#include <numeric>
+#include <string>
+
+namespace vpd
+{
+
+/* @brief Keyword VPD parser function*/
+std::variant<types::ParsedVPD, types::KeywordVpdMap> KeywordVpdParser::parse()
+{
+    try
+    {
+    m_vpdIterator = m_keywordVpdVector.begin();
+
+    if(*m_vpdIterator != constants::KW_VPD_START_TAG)
+    {
+        throw(DataException("Invalid Large resource type Identifier String"));
+    }
+
+    checkNextBytesValidity(sizeof(constants::KW_VPD_START_TAG));
+    std::advance(m_vpdIterator, sizeof(constants::KW_VPD_START_TAG));
+
+    uint16_t dataSize = getKwDataSize();
+
+    checkNextBytesValidity(constants::TWO_BYTES + dataSize);
+    std::advance(m_vpdIterator, constants::TWO_BYTES + dataSize);
+
+    // Check for invalid vendor defined large resource type
+    if (*m_vpdIterator != constants::KW_VPD_START_TAG)
```

Comment:
We already have this check at line number 20. Do we need this here again?

Reply:
I'll  cross check.

Reply:
1st Time we check for keyword_vpd TAG. (KW_VPD_START_TAG)
At this position we have to check for TAG which indicates   keyword-value pairs starting now. (KW_VPD_PAIR_START_TAG)

Reply:
So that needs a change? Because right now I see same tag being used at both the palces.